File: Codigos AEDS2/TP1/Palindromo.c
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <stdio.h>

bool isfim(char *string){
    return ( string[0] == 'F' && string[1] == 'I' && string[2] == 'M');
}

bool Palindromo(char *string ){
    bool eh = true;
    int tamanho = strlen(string)-1;

    for (int i = 0; i < tamanho/2 ; i++)
    {
        if ( string[i] != string[tamanho-i-1] )
        {
            eh = false;
            i = tamanho;
        }
    }   
    return eh;
}

int main(){
    char Entrada[1000];
    bool palindromo;
    
    fgets( Entrada , 1000 , stdin);

    while( !isfim(Entrada) )
    {
        palindromo = Palindromo(Entrada);
        
        if (palindromo)
        {
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }
       
        fgets(Entrada , 1000 , stdin );        
        
    }  
    
    return 0;
}

File: Codigos AEDS2/TP1/Palindromo.java
class Palindromo{
    public static boolean isFim(String s){
        return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    static boolean isPalindromo(String s){
        
        int length = s.length();

        if (length < 2) // verifica se a string está vazia
            return true;
        else {
        // Chega os lados opostos da string
            if (s.charAt(0) != s.charAt(length - 1))
                return false;
            // cahamda de retorno
            else
                return isPalindromo(s.substring(1, length - 1));
        }
    }

    public static void main(String[] args) {

        String[] entrada = new String[1000];

        do {
            entrada[0] = MyIO.readLine();
            if(isPalindromo(entrada[0])){
                System.out.println("SIM");
            }else{
                System.out.println("NAO");
            }
        } while (isFim(entrada[0]) == false);
           //Desconsiderar ultima linha contendo a palavra FIM
    
    }
}
